### PR TITLE
Changes how explosive guardian bombs work.

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

Explosive holoparsites now make their targets explode by listening to signals, rather than putting the rigged object into another object.

## Why It's Good For The Game

It's cleaner and handles stuff like overlays or other functionality much better.

## Changelog
:cl:
tweak: Slightly changed how explosive holoparasite bombs work. Actual gameplay impact should be minimal.
balance: The distance requirement for finding bombs on examine has been removed.
/:cl:

### CODE REVIEW

**Comment by XDTM** [comment]
what happens if a bomb is triggered after the holoparasite dies?

**Comment by nemvar** [comment]
The bomb dies with the holoparasite.

File: code/modules/mob/living/simple_animal/guardian/types/explosive.dm [hunk 1]
```diff
@@ -38,61 +39,32 @@
...
-/obj/guardian_bomb/proc/detonate(mob/living/user)
-	if(isliving(user))
-		if(user != spawner && user != spawner.summoner && !spawner.hasmatchingsummoner(user))
-			to_chat(user, "<span class='danger'><B>[src] was boobytrapped!</B></span>")
-			to_chat(spawner, "<span class='danger'><B>Success! Your trap caught [user]</B></span>")
-			var/turf/T = get_turf(src)
-			stored_obj.forceMove(T)
+/mob/living/simple_animal/hostile/guardian/bomb/proc/kaboom(atom/source, mob/living/explodee)
+	if(istype(explodee))
+		if(explodee != src && explodee != summoner && !hasmatchingsummoner(explodee))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/guardian/types/explosive.dm]:
```suggestion
		if(explodee == src || explodee == summoner || hasmatchingsummoner(explodee))
		    return
		//rest of code here
```

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/guardian/types/explosive.dm]: actually

```DM
if(!istype(explodee))
    return
if(explodee == src || explodee == summoner || hasmatchingsummoner(explodee))
    return

//rest of code here
```

File: code/modules/mob/living/simple_animal/guardian/types/explosive.dm [hunk 2]
```diff
@@ -38,61 +39,32 @@
...
-/obj/guardian_bomb/attack_hand(mob/living/user)
-	detonate(user)
+/mob/living/simple_animal/hostile/guardian/bomb/proc/display_examine(datum/source, mob/user, text)
+	text += "<span class='holoparasite'>It glows with a strange <font color=\"[namedatum.colour]\">light</font>!</span>"
 
-/obj/guardian_bomb/examine(mob/user)
-	. = stored_obj.examine(user)
-	if(get_dist(user,src)<=2)
-		. += "<span class='holoparasite'>It glows with a strange <font color=\"[spawner.namedatum.colour]\">light</font>!</span>"
+/mob/living/simple_animal/hostile/guardian/bomb/proc/RemoveSignals(datum/A)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/guardian/types/explosive.dm]: this should prob be a define instead of a proc that just calls procs.

**Pull request merged**